lib/LaTeXML/Package/graphicx.sty.ltxml

```diff
@@ -47,16 +48,24 @@ DefMacro('\includegraphics OptionalMatch:* []',
 
 DefConstructor('\@includegraphicx OptionalMatch:* OptionalKeyVals:Gin Semiverbatim',
   "<ltx:graphics graphic='#path' candidates='#candidates' options='#options'/>",
-  alias      => '\includegraphics',
-  sizer      => \&image_graphicx_sizer,
-  scope      => 'global',
+  alias          => '\includegraphics',
+  sizer          => \&image_graphicx_sizer,
+  scope          => 'global',
+  afterConstruct => sub {
+    my ($document, $whatsit) = @_;
+    my $alt = $whatsit->getProperty('alt');
+    # Trickery to set @description EVEN IF empty string (constructor shorthand omits it)
+    if (defined $alt) {
+      $document->getNode->lastChild->setAttribute(description => ToString($alt)); } },
```

Comment:
Cool, this works great for `alt={}` and `alt=\relax`.

I wonder if we can also default to empty for some expected uses of quotes. And just generally sanitize away unfortunate uses of outer-wrapping quotes that would make a screen reader speak unnecessary syntax.

I got some mileage with a couple of extra regexes:
```perl
    # Trickery to set @description EVEN IF empty string (constructor shorthand omits it)
    if (defined $alt) {
      my $description = ToString($alt);
      # sanitize away accidental quotations/fencing wrappers
      $description =~ s/^\W+//;
      $description =~ s/\W+$//;
      # discard values that do not contain traces of language
      $description = '' if $description !~ /\w/;
      $document->getNode->lastChild->setAttribute(description => $description); } },
```

which seems to produce nicely sanitized values for:
```tex
\includegraphics[alt="",width=100pt]{fig1.pdf}

\includegraphics[alt='',width=100pt]{fig1.pdf}

\includegraphics[alt={},width=100pt]{fig1.pdf}

\includegraphics[alt=first order approximation,width=100pt]{fig1.pdf}

\includegraphics[alt={second order approximation of the first kind},width=100pt]{fig1.pdf}

\includegraphics[alt="third order approximation of the second kind.",width=100pt]{fig1.pdf}

\includegraphics[alt=\relax,width=100pt]{fig1.pdf}

\includegraphics[alt,width=100pt]{fig1.pdf} % also needs the empty default in DefKeyVal
```

I realize that some of these may be considered "bad LaTeX" keyvals, but even so I expect people will end up writing them on occasion. So might as well guard some of the obvious cases.

Reply:
Not so much "bad" TeX, as just not TeX, and very unexpected by a TeX author.  Besides, your regexps would strip away *all* leading/trailing non-alphanumerics.

Reply:
Further point (in support of no regexps), for forward compatibility: it is conceivable that graphicx will implement `alt` in the future, so the semantics implemented by LaTeXML should be as simple as possible.

Reply:
@xworld21 oh, it is already implemented in graphicx, see [ltnews34, page 6](https://www.latex-project.org/news/latex2e-news/ltnews34.pdf) which also links to the github issue.

Maybe I am being overly cautious due to my experiences in dealing with arXiv sources, but I am indeed thinking about forward compatibility - the error we have if an author used a simple boolean `alt` are not informative, and if an author mistakenly used quotation marks (with or without content), they're likely to not notice until a non-sighted reader encounters the bumpy speech in their screen reader.

Whatever Bruce decides is fine by me here, just an extra consideration on my part.

Reply:
If they don't give a value to alt, they'll get the same message they always get for missing keyval values (whether or not it is informative).  And besides, it just doesn't make any sense.

And if they intentionally used quotation marks (or [], (), math, or any other non alphanumeric), and we silently remove them? What if they used the wrong words and didn't proofread?  If the eventual LaTeX3 definition and usage suggests that the value should be stripped or massaged in some particular way, we can of course follow that usage.  But to proactively choose a very un-TeX-like interpretation? Nah.